libs/sdk-core/src/breez_services.rs

```diff
@@ -297,10 +297,25 @@ impl BreezServices {
             }
             ValidatedCallbackResponse::EndpointSuccess { data: cb } => {
                 let pay_req = SendPaymentRequest {
-                    bolt11: cb.pr,
+                    bolt11: cb.pr.clone(),
                     amount_msat: None,
                 };
-                let payment = self.send_payment(pay_req).await?.payment;
+
+                let payment = match self.send_payment(pay_req).await {
+                    Ok(p) => Ok(p),
+                    Err(SdkError::SendPaymentFailed { err }) => {
+                        let invoice = parse_invoice(cb.pr.as_str())?;
+
+                        return Ok(LnUrlPayResult::PayError {
```

Comment:
Two points here:

- for consistency, I would remove the `return ...;` and just place the `Ok(..)` on the last line of the block, just like the other two branches of `match`
- is this really an `Ok(..)` and not an `Err(..)`?

Reply:
I think this needs to stay as is.

- The return is explicitly required as we want to exit the function and return the `PayError` to the caller.
- It must be an `Ok(..)` instead of an `Err(..)` as this is the current way the method handles errors, as pointed out by @roeierez https://github.com/breez/breez-sdk/pull/550#discussion_r1371648746.

Reply:
> The return is explicitly required as we want to exit the function and return the `PayError` to the caller.

We unwrap the result (and throw this error as well, if it would be wrapped as `Err`) just a few lines below in L317.

> It must be an `Ok(..)` instead of an `Err(..)` as this is the current way the method handles errors, as pointed out by @roeierez [Add payment hash to `lnurl_pay()` return type #550 (comment)](https://github.com/breez/breez-sdk/pull/550#discussion_r1371648746).

- The current way (before the PR): we are not mixing success and error in the success case
- The current way (in this PR): we are mixing them with the `Ok(..)` I mentioned above

So changing the `Ok(..)` into an `Err(..)` would address Roei's point around mixing result types. It would also make future migrations to `SdkError` simpler IMO.

Reply:
Nevermind, my bad.

You want to return `LnUrlPayResult` in the success case, so `Ok(..)` is correct.

The generic `anyhow::Error` thrown now only takes a string as arg, so cannot be used to wrap anything more meaningful. The `send_payment` step already throws an `SdkError`, but that's converted back to `anyhow::Error` when unwrapping on L137.

All this can be handled better with `SdkError` in the future.